### ISSUE TITLE
error unknown plugin 

### ISSUE THREAD

**Issue by TonioDF** [PascalMinder/geoblock#31]
Hello, I am new to traefik.  I can't get this plugin to work.  I follow the installation process it indicates, but then when I try to push it into a docker instance, I get this error:
"entryPointName":"http","level":"error","msg":"plugin: unknown plugin type: geoblock"
When I carry out the installation process, everything seems to go well (I leave a screenshot) I get the error when I want to apply it in a docker container
![Screenshot_2023-01-15-16-31-16-43_3aea4af51f236e4932235fdada7d1643](https://user-images.githubusercontent.com/91021876/212550913-7ccb5754-ca9f-4b48-a156-875a4d9c56d8.jpg)

In the labels of my docker containers I add this:
traefik.http.routers."servicedocker".middlewares=my-geoblock@file

Thanks

**Comment by PascalMinder** [comment]
Did you check out the example [docker setup](https://github.com/PascalMinder/geoblock/tree/main/docker)?

Can you also post your configuration files `traefik.yml` and `dynamic-configuration.yml`? And also perhaps your `docker-compose` file?

**Comment by TonioDF** [comment]
My traefik.yml :

`global:
  checkNewVersion: true
  sendAnonymousUsage: true

serversTransport:
  insecureSkipVerify: true

entryPoints:
  # Not used in apps, but redirect everything from HTTP to HTTPS
  http:
    address: :80
    http:
      redirections:
        entryPoint:
          to: https
          scheme: https

  # HTTPS endpoint, with domain wildcard
  https:
    address: :443
    http:
      tls:
        # Generate a wildcard domain certificate
        certResolver: letsencrypt
        domains:
          - main: xxx.duckdns.org  
            sans:
              - '*.xxx.duckdns.org'  
      middlewares:
        - securityHeaders@file
providers:
  providersThrottleDuration: 2s

  # File provider for connecting things tat are outside of docker / defining middleware
  file:
    filename: /etc/traefik/fileConfig.yml
    watch: true
# Docker provider for connecting all apps that are inside of the docker network
  docker:
    watch: true
    network: proxy # Add Your Docker Network Name Here
    # Default host rule to containername.domain.example
    defaultRule: "Host(`{{ index .Labels \"com.docker.compose.service\"}}.xxx.duckdns.org`)" 
    swarmModeRefreshSeconds: 15s
    exposedByDefault: false

# Enable traefik ui
api:
  dashboard: true
  insecure: false

# Log level INFO|DEBUG|ERROR
log:
  filePath: "etc/traefik.log"
  level: ERROR
  format: json


# Use letsencrypt to generate ssl serficiates
certificatesResolvers:
  letsencrypt:
    acme:
      email: xxx
      storage: /etc/traefik/acme.json
      dnsChallenge:
        provider: duckdns
        # Used to make sure the dns challenge is propagated to the rights dns servers
        resolvers:
          - "1.1.1.1:53"
          - "8.8.8.8:53"

# Plugins
experimental:
  plugins:  
    geoblock:
      moduleName: "github.com/PascalMinder/geoblock"
      version: "v0.2.4"
`
My dynamic-config.yml :

`http:
  routers:
    #añadir seguridad al dashboard
    dashboard:
      entryPoints:
        - https
      rule: Host(`traefik.xxx.duckdns.org`) && (PathPrefix(`/api`) || PathPrefix(`/dashboard`))
      service: api@internal
      middlewares:
        - auth
  middlewares:
    auth:
      basicAuth:
        users:
          - "xxx"
    
    securityHeaders:
      headers:
        customResponseHeaders:
          X-Robots-Tag: "none,noarchive,nosnippet,notranslate,noimageindex"
          server: ""
        sslProxyHeaders:
          X-Forwarded-Proto: https
        referrerPolicy: "same-origin"
        hostsProxyHeaders:
          - "X-Forwarded-Host"
        customRequestHeaders:
          X-Forwarded-Proto: "https"
        customResponseHeaders:
          X-Forwarded-Proto: "https"
        contentTypeNosniff: true
        browserXssFilter: true
        forceSTSHeader: true
        stsIncludeSubdomains: true
        stsSeconds: 63072000
        stsPreload: true

    my-GeoBlock:
      plugin:
        GeoBlock:
          allowLocalRequests: true
          logLocalRequests: false
          logAllowedRequests: false
          logApiRequests: false
          api: "https://get.geojs.io/v1/ip/country/{ip}"
          apiTimeoutMs: 500
          cacheSize: 25
          forceMonthlyUpdate: true
          allowUnknownCountries: false
          unknownCountryApiResponse: "nil"
          backListMode: false
          countries:
            - ES
            - PT
`
My traefik  docker-compose.yml :

`version: '3'
services:
  traefik:
    container_name: traefik
    image: traefik:2.6
    ports:
      - 80:80
      - 443:443
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./config/:/etc/traefik/
      - ./traefik.log:/etc/traefik.log
    networks:
      - proxy 
    labels:
      traefik.http.routers.api.rule: Host(`xxx.xxx.duckdns.org`)    
      traefik.http.routers.api.service: api@internal    
      traefik.enable: true   
    environment:
      DUCKDNS_TOKEN: 
    restart: unless-stopped
networks:
  proxy:    
    external: true
`
Example of a webdav container configuration where I add the label my-geoblock .
My docker-compose.traefik.yml:

`version: '3'
services:
  webdav:
    networks:
      - proxy
    labels:
      - traefik.enable=true
      - traefik.http.services.webdav.loadbalancer.server.port=8080
      - traefik.http.routers.webdav.entrypoints=https
      - traefik.http.routers.webdav.rule=Host(`${FQDN}`)
      - traefik.http.middlewares.webdav-https-redirect.redirectscheme.scheme=https
      - traefik.http.routers.webdav.middlewares=webdav-https-redirect
      - traefik.http.routers.webdav-secure.entrypoints=https
      - traefik.http.routers.webdav-secure.rule=Host(`${FQDN}`)
      - traefik.http.routers.webdav-secure.tls=true
      - traefik.http.routers.webdav-secure.tls.certresolver=letsencrypt
 #####   when i add this argument is when i get the error   #####
      - traefik.http.routers.webdav.middlewares=my-GeoBlock@file
`

I know that in all probability it is because of a bad configuration of mine, i have also read the documentation and have done many tests but all without success.

Thank you very much for your answer and your interest, I am very sorry to bother you

**Comment by PascalMinder** [comment]
I'm not sure, but in your `traefik.yml`:
```
experimental:
  plugins:  
    geoblock:            <-----------
```

and in your `dynamic-configuration.yml` you wrote:
```
    my-GeoBlock:
      plugin:
        GeoBlock:         <---------
```

I think you should write both in small caps.

**Comment by TonioDF** [comment]
> 

I've already tried it, and it still doesn't work.

![Screenshot_2023-01-17-21-52-34-88_3aea4af51f236e4932235fdada7d1643](https://user-images.githubusercontent.com/91021876/213009949-5940d193-2166-4906-832b-a11ffff99933.jpg)

I've been thinking about this problem for several days and I can't solve it.
Thanks @PascalMinder

**Comment by TonioDF** [comment]
Resolved !!! The problem was with version 2.6 of docker compose, I updated to 2.9.6 and it worked.
Thanks @PascalMinder
